soccer/MainWindow.cpp

```diff
@@ -974,30 +982,44 @@ void MainWindow::channel(int n) {
 // Simulator commands
 
 void MainWindow::on_actionCenterBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_pos()->set_x(0);
-    cmd.mutable_ball_pos()->set_y(0);
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    ball_replace->set_x(0);
+    ball_replace->set_y(0);
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionStopBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    Geometry2d::Point ballPos =
+        _ui.fieldView->getTeamToWorld() * state()->ball.pos;
+    ball_replace->set_x(ballPos.x());
+    ball_replace->set_y(ballPos.y());
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionResetField_triggered() {
```

Comment:
The API provided by grSim doesn't give us the ability to just "call" that function. I see two options here:
1. We add a new grSim protobuf packet that gives us the option for reset which makes us diverge even further from grSim.
2. We just hardcode robot positions and send a series of RobotReplacement messages.
```protobuf
message grSim_RobotReplacement {
required double x=1;
required double y=2;
required double dir=3;
required uint32 id=4;
required bool yellowteam=5;
}
```

Thoughts?